docs/overview.md

```diff
@@ -74,9 +74,34 @@ Comparison                      | `>`, `<`, `>=`, `<=`
 Boolean operations              | `!`, `&&`, <code>&#124;&#124;</code>
 Reference equality              | `===`, `!==`
 Structural equality             | `==`, `!=`
+
+## Implicit Return
+
+There is no `return` keyword in Reason. The last expression in a block or
+function definition is the return value for that block.
```

Comment:
I don't think that "return values" make sense for blocks, only for functions. Blocks are expressions that evaluate to a value like every other expression, so it's probably more accurate to say that "The last item in an expression block is what that expression block evaluates to". 

Then somehow explain that functions just have a single expression on the right hand side of their `=>`, and if it is a block so be it - and that expression will evaluate like every other expression, with no ability to early return.

Reply:
Changed the example to deal with functions so this is less confusing.